lib/fastlane/plugin/browserstack/actions/upload_to_browserstack_app_automate_action.rb

```diff
@@ -0,0 +1,105 @@
+require 'fastlane/action'
+require_relative '../helper/browserstack_helper'
+require 'json'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      BROWSERSTACK_APP_ID ||= :BROWSERSTACK_APP_ID
+    end
+    class UploadToBrowserstackAppAutomateAction < Action
+      SUPPORTED_FILE_EXTENSIONS = ["apk", "ipa", "zip"]
+      UPLOAD_API_ENDPOINT = "https://api-cloud.browserstack.com/app-automate/upload"
+
+      def self.run(params)
+        browserstack_username = params[:browserstack_username] # Required
+        browserstack_access_key = params[:browserstack_access_key] # Required
+        file_path = params[:file_path] # Required
+
+        UI.message("Uploading app to BrowserStack AppAutomate...")
+
+        browserstack_app_id = Helper::BrowserstackHelper.upload_file(browserstack_username, browserstack_access_key, file_path, UPLOAD_API_ENDPOINT)
+
+        # Set 'BROWSERSTACK_APP_ID' environment variable, if app upload was successful.
+        ENV['BROWSERSTACK_APP_ID'] = browserstack_app_id
+
+        UI.success("Successfully uploaded app " + file_path + " to BrowserStack AppAutomate with app_url : " + browserstack_app_id)
+
+        UI.success("Setting Environment variable BROWSERSTACK_APP_ID = " + browserstack_app_id)
+
+        # Setting app id in SharedValues, which can be used by other fastlane actions.
+        Actions.lane_context[SharedValues::BROWSERSTACK_APP_ID] = browserstack_app_id
+      end
+
+      # Validate file_path.
+      def self.validate_file_path(file_path)
+        UI.user_error!("No file found at '#{file_path}'.") unless File.exist?(file_path)
+
+        # Validate file extension.
+        file_path_parts = file_path.split(".")
+        unless file_path_parts.length == 2 && SUPPORTED_FILE_EXTENSIONS.include?(file_path_parts[1])
```

Comment:
There is no restriction on the number of `.`s that can be there in a filename. Use the last part to verify file extension.

Reply:
Done